### ISSUE TITLE
Align formatting (bold, emph, etc.) with djot AST

### ISSUE THREAD

**Issue by bdarcus** [bdarcus/csl-next#30]
@jgm, when you get a chance, curious what you suggest on integrating djot.js here, for both subfield formatting, and ultimately to integrate it into djot document processing.

Edit: thinking about this more, may need to both use djot.parse on the fields and work with the AST otherwise? Basically, I'm thinking of adding a `toDjotAST` method.

But if I need to transform content to the AST, what's the best way to do that? Any suggestions on where to look in `djot.js` for inspiration?

Note to self: the djot playground includes [filters](https://djot.net/playground/)!

Also: https://mfrachet.github.io/create-frontend-framework/templating/template-literals.html#abstracting-for-others-html-elements

------------------

As part of the more general #5.

Templates (aka macros in 1.0) look like this:

```yaml
- variable: author
  emph: true
  wrap: parentheses
```

```js
[
  {
    "variable": "author",
    "emph": true,
    "wrap": "parentheses"
  }
]
```

Basically, better to convert this to:

1. The Djot AST, or ...
2. The Djot string markup, and let `djot.parse()` handle the rest?

In 1, then, I'd run `djot.parse()` first, and in 2 last.

I know I can just create the AST from that, and then insert subfield content as `children`.

But one issue might be that this CSL model here is flat, while Djot is pretty hierarchical.

So with this, the emph is contained  in that inner child object.

```js
> djot.parse("hello *_there_*").children[0];
{
  tag: "para",
  children: [
    { tag: "str", text: "hello ", pos: undefined },
    {
      tag: "strong",
      children: [ { tag: "emph", children: [Array], pos: undefined } ],
      pos: undefined
    }
  ],
  attributes: undefined,
  pos: undefined
}
```
